kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaCommandDecoder.java

```diff
@@ -309,7 +309,19 @@ protected void writeAndFlushResponseToClient(Channel channel) {
                 ByteBuf result = responseToByteBuf(response.getResponseFuture().get(
                         kafkaConfig.getRequestTimeoutMs(),
                         TimeUnit.MILLISECONDS), response.getRequest());
-                channel.writeAndFlush(result);
+                channel.writeAndFlush(result)
+                        .addListener(future -> {
+                            if (future.isDone()) {
```

Comment:
The check of `future.isDone()` is redundant, see  https://netty.io/4.1/api/io/netty/channel/ChannelFuture.html#addListener-io.netty.util.concurrent.GenericFutureListener-

>  The specified listener is notified when this future is done.